### PR TITLE
Scavengers: Expand the functionality of /scav timer

### PR DESCRIPTION
**Pull request by aQrator**
Adjust the _/scav timer_ function to allow for turning off the timer, and to enable fractional times once again.

### CODE REVIEW

**Comment by aQrator** [comment]
Proof of the request by the RO LittEleven
afbeelding

File: server/chat-plugins/scavengers.ts [hunk 1]
```diff
@@ -577,7 +577,10 @@ export class ScavengerHunt extends Rooms.RoomGame {
 			this.timer = null;
 			this.timerEnd = null;
 		}
-
+		
+		if (minutes == 0) { 
+			return 'off';
+		}
```
Comment (AnnikaCodes) [inline server/chat-plugins/scavengers.ts:583]:
```suggestion

		if (minutes === 0) { 
			return 'off';
		}		
```

File: server/chat-plugins/scavengers.ts [hunk 2]
```diff
@@ -1483,9 +1486,16 @@ const ScavengerCommands: ChatCommands = {
...
 		const game = room.getGame(ScavengerHunt);
 		if (!game) return this.errorReply(`There is no scavenger hunt currently running.`);
+		
+		let minutes;
+		if (target == "off") {
+			minutes = 0;
+		}
+		else {
+			minutes = parseFloat(target);
+		}
```
Comment (AnnikaCodes) [inline server/chat-plugins/scavengers.ts:1496]:
```suggestion

		const minutes = (toID(target) === 'off' ? 0 : parseFloat(target));
```

Comment (AnnikaCodes) [reply server/chat-plugins/scavengers.ts:1496]: (To be clear, `game.setTimer(0)` should clear the timeout, not set it to 0 milliseconds, right?)

**Review by AnnikaCodes — CHANGES_REQUESTED** [review]

File: server/chat-plugins/scavengers.ts (hunk 2 above)
Comment (aQrator) [reply server/chat-plugins/scavengers.ts:1496]: That is correct

Comment (AnnikaCodes) [reply server/chat-plugins/scavengers.ts]: Also, if you were to do this with an if statement, I'd do it like this:
```ts
let minutes = 0;
if (target !== 'off) minutes = parseFloat(target);
```
or
```ts
let minutes;
if (target === 'off') {
     minutes = 0;
} else {
     minutes = parseFloat(target);
}
```

File: server/chat-plugins/scavengers.ts (hunk 1 above)
Comment (aQrator) [reply server/chat-plugins/scavengers.ts]: Adjusted alongside other errors

File: server/chat-plugins/scavengers.ts (hunk 2 above)
Comment (aQrator) [reply server/chat-plugins/scavengers.ts]: Thanks. I will remember this in the future. For now, I've just used the 1-line const declaration

**Review by AnnikaCodes — APPROVED** [review]
Looks fine to me

**Comment by aQrator** [comment]
Thank you. When can we expect this update to go live?

**Comment by AnnikaCodes** [comment]
> Thank you. When can we expect this update to go live?

idt I'm supposed to merge things, so give it a few days so that Zarel (or xfix) can review, and merge and someone can hotpatch the change in

**Comment by sparkychildcharlie** [comment]
looks good

**Review by xfix — APPROVED** [review]

**Pull request merged**